src/toil/cwl/cwltoil.py

```diff
@@ -796,7 +801,8 @@ def run(self, fileStore):
 
         outobj = {}
         for out in self.cwlwf.tool["outputs"]:
-            outobj[shortname(out["id"])] = (shortname(out["outputSource"]), promises[out["outputSource"]].rv())
+            src = simplify_list(out["outputSource"])
+            outobj[shortname(out["id"])] = (shortname(src), promises[src].rv())
```

Comment:
I think this was fixed by https://github.com/BD2KGenomics/toil/pull/2069 ; if so then I prefer that solution as it ensures all contributing links are processed correctly. If you have a complete example that doesn't work with that PR please share it so I can evaluate this in conjunction